go2ts_test.go

```diff
@@ -159,9 +165,46 @@ export interface ADifferentName {
 	assert.Equal(t, expected, b.String())
 }
 
-func TestAdd_NonStructType_ReturnsError(t *testing.T) {
+func TestRender_NonStructTypes_Success(t *testing.T) {
+	go2ts := New()
+	err := go2ts.Add(Data{})
+	require.NoError(t, err)
+	var b bytes.Buffer
+	err = go2ts.Render(&b)
+	require.NoError(t, err)
+	expected := `// DO NOT EDIT. This file is automatically generated.
+
+export type Data = { [key: string]: any };
+`
+	assert.Equal(t, expected, b.String())
+}
+
+func TestRender_NonStructTypeWithName_Success(t *testing.T) {
+	go2ts := New()
+	err := go2ts.AddWithName(Data{}, "SomeNewName")
+	require.NoError(t, err)
+	var b bytes.Buffer
+	err = go2ts.Render(&b)
+	require.NoError(t, err)
+	expected := `// DO NOT EDIT. This file is automatically generated.
+
+export type SomeNewName = { [key: string]: any };
+`
+	assert.Equal(t, expected, b.String())
+}
+
+type HasUnsupportedFieldTypes struct {
+	C complex128
+}
+
+func TestAdd_UnsupportedType_Panic(t *testing.T) {
+	defer func() {
+		if r := recover(); r == nil {
+			assert.Fail(t, "Complex128 is unsupported and should panic.")
+		}
+	}()
+
 	go2ts := New()
-	err := go2ts.Add("this is not a struct")
-	assert.Error(t, err)
-	assert.Equal(t, "string is not a struct.", err.Error())
+	err := go2ts.Add(HasUnsupportedFieldTypes{})
```

Comment:
Might be clearer with `assert.PanicsWithError`[1], e.g.:

```
assert.PanicsWithError(t, "Go Kind Complex128 cannot be converted to TypeScript.", func(){
  go2ts.Add(HasUnsupportedFieldTypes{})
})
```

[1] https://godoc.org/github.com/stretchr/testify/assert#PanicsWithError

Reply:
I couldn't get that to work, didn't fiddle with it too long, keeping this the same.